spec/indexers/facet_indexer_spec.rb

```diff
@@ -30,4 +30,36 @@
       end
     end
   end
+
+  context "when the resource has structure" do
+    it "indexes its presence" do
+      file_set = FactoryBot.create_for_repository(:file_set)
+      scanned_resource = FactoryBot.create_for_repository(
+        :scanned_resource,
+        member_ids: file_set.id,
+        thumbnail_id: file_set.id,
+        logical_structure: [
+          { label: "testing", nodes: [{ label: "Chapter 1", nodes: [{ proxy: file_set.id }] }] }
+        ]
+      )
+
+      output = described_class.new(resource: scanned_resource).to_solr
+      expect(output[:has_structure_bsi]).to be true
+    end
+  end
+
+  context "when the resource does not have structure" do
+    it "indexes its absence" do
+      file_set = FactoryBot.create_for_repository(:file_set)
+      scanned_resource = FactoryBot.create_for_repository(
+        :scanned_resource,
+        member_ids: file_set.id,
+        thumbnail_id: file_set.id,
+        logical_structure: []
```

Comment:
Does this test still pass if you don't set this? I want to make sure `logical_structure` is never nil, because the indexer would throw an error.

Reply:
The test passes even if there's no logical_structure (or the object doesn't respond to `logical_structure`) — I started out doing a bunch of checking to make sure to avoid errors, but wound up just doing an unqualified `rescue` to make it simpler and catch any error that might happen: https://github.com/pulibrary/figgy/pull/2793/files#diff-2adfa7e07f80ce4d413250621142b804R47